test/Makefile

```diff
@@ -1,8 +1,13 @@
 ARCH ?= x86_64
+# in the docker image, CC may be set to cc but we need clang
+ifdef CC
```

Comment:
Isn't CC ?= clang-6.0 effectively the same as doing the ifdef CC?

Reply:
My understanding is that this will set `CC = clang-6.0` no matter what, which is probably not what we want given the readme example. This should probably just be `CC ?= clang-6.0`.

Reply:
If I try to remove the `ifdef` and just put ` CC ?= clang-6.0`, it works fine on a "native" rhel8, but when I try the "standard" `docker-compose run --rm test-builder` build on a Mac, `CC` is set to `cc` (due to ttps://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html ) and I get the error below [1].  How about if me use the make's `--no-builtin-variables` option (equivalent to `-R`) to prevent `CC`  from being set to `cc`?  That does work on Mac with `docker-compose`. Added it in 46afa04 and removed `ifdef` - let me know what you think.

[1]
```
$ docker-compose run --rm test-builder
Creating test_test-builder_run ... done
CC: cc
KERNEL_HEADER_VERSION: 4.4.0-98-generic
KERNEL_HEADERS_ROOT: /usr/src/linux-headers-4.4.0-98-generic
rm -rf test_program_*
cc -target x86_64 -D__KERNEL__ -D__BPF_TRACING__ -Wunused -Wall -Werror -Wno-pointer-sign -Wno-address-of-packed-member -Wno-compare-distinct-pointer-types -Wno-gnu-variable-sized-type-not-at-end -Wno-macro-redefined -Wno-sometimes-uninitialized -Wno-tautological-compare -fno-stack-protector -Xclang -disable-llvm-passes -O2 -D__ASM_SYSREG_H -D__TARGET_ARCH_x86 -emit-llvm -c test_program.c -I src/ -I /usr/src/linux-headers-4.4.0-98-generic/arch/x86/include -I /usr/src/linux-headers-4.4.0-98-generic/arch/x86/include/uapi -I /usr/src/linux-headers-4.4.0-98-generic/arch/x86/include/generated -I /usr/src/linux-headers-4.4.0-98-generic/arch/x86/include/generated/uapi -I /usr/src/linux-headers-4.4.0-98-generic/include -I /usr/src/linux-headers-4.4.0-98-generic/include/uapi -I /usr/src/linux-headers-4.4.0-98-generic/include/generated -I /usr/src/linux-headers-4.4.0-98-generic/include/generated/uapi -o - | \
	opt-6.0 -O2 -mtriple=bpf-pc-linux | llvm-dis-6.0 | \
	llc-6.0 -march=bpf -filetype=obj -o test_program_x86_64
/bin/sh: 1: cc: not found
```

Reply:
Good catch, I like the `--no-builtin-variables` better. We'll merge that in.

Thanks for contributing!